### PR TITLE
Do not use bare 'except:': flake8 --select=E722

### PR DESCRIPTION
**Pull request by cclauss**
Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on. Prefer `except Exception:`. If you’re sure what you’re doing, be explicit and write `except BaseException:`.

https://realpython.com/the-most-diabolical-python-antipattern/
https://www.python.org/dev/peps/pep-0008/

### CODE REVIEW

**Review by jtbandes — APPROVED** [review]

File: rosbridge_library/test/capabilities/test_subscribe.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test_update_params(self):
             self.assertEqual(subscription.queue_length, min_queue_length)
             self.assertEqual(subscription.fragment_size, min_frag_size)
             self.assertEqual(subscription.compression, "png")
-        except:
+        except:  # noqa: E722  # Will unregister and raise
             subscription.unregister()
             raise
```
Comment (jtbandes) [inline rosbridge_library/test/capabilities/test_subscribe.py]: Should these use `finally:` instead of `except: ... raise`?

Comment (cclauss) [reply rosbridge_library/test/capabilities/test_subscribe.py]: I think not...

 % `python3`
```python
>>> try:
...     print("Hello")
... finally:
...     raise
...
Hello
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
RuntimeError: No active exception to reraise
```
We want the exception to propagate up the call stack.

Comment (jtbandes) [reply rosbridge_library/test/capabilities/test_subscribe.py]: sorry, I meant 
```py
try:
  ...
finally:
  subscription.unregister()
```

Comment (cclauss) [reply rosbridge_library/test/capabilities/test_subscribe.py]: ~I think we are good the way it is.  If failure unregister and raise else unregister (the line after) does what we want.~

Comment (cclauss) [reply rosbridge_library/test/capabilities/test_subscribe.py]: Oh...
```python
try:
    xyz
except:  # noqa: E722  # Will raise
    raise
finally:
    subscription.unregister()
```

Comment (jtbandes) [reply rosbridge_library/test/capabilities/test_subscribe.py]: Is the `except: raise` part necessary? I'm suggesting that you can avoid the `# noqa` by omitting the except clause and only using finally.

Comment (cclauss) [reply rosbridge_library/test/capabilities/test_subscribe.py]:
```
>>> try:
...     Hello
... finally:
...     print("Goodby")
...
Goodby
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
NameError: name 'Hello' is not defined
```

**Pull request merged**